### PR TITLE
feat: add checkbox component

### PR DESCRIPTION
**Pull request by hhochart29**

### CODE REVIEW

File: packages/mikado_reborn/src/components/Checkbox/Checkbox.scss [hunk 1]
```diff
@@ -0,0 +1,56 @@
+@import '../../assets/styles/styles.scss';
+
+.mkr__checkbox {
+
+  background-color: map-get($colors, 'white');
+  border: 1px solid map-get($colors, 'neutral-20');
+  border-radius: 1rem;
```
Comment (GuillaumeBAECHLER) [inline packages/mikado_reborn/src/components/Checkbox/Checkbox.scss:7]:
```suggestion
  border-radius: 4px;
```

File: packages/mikado_reborn/src/components/Checkbox/Checkbox.scss [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+  &.medium {
+    width: 3rem;
+    height: 3rem;
+    border-radius: 6px;
+  }
+
+  &:focus {
+    outline: none;
+    box-shadow: 0 0 0 0.5rem transparentize(map-get($colors, 'secondary'), 0.8);
```
Comment (GuillaumeBAECHLER) [inline packages/mikado_reborn/src/components/Checkbox/Checkbox.scss:17]: Can you use `$primary-secondary-20` in the sass tokens and make a `secondary-20` variable in color files please ? This one is missing but should be used here.

File: packages/mikado_reborn/src/components/Checkbox/Checkbox.scss [hunk 3]
```diff
@@ -0,0 +1,56 @@
+@import '../../assets/styles/styles.scss';
+
+.mkr__checkbox {
+
```
Comment (GuillaumeBAECHLER) [inline packages/mikado_reborn/src/components/Checkbox/Checkbox.scss:4]:
```suggestion
```

File: packages/mikado_reborn/src/components/Checkbox/Checkbox.vue [hunk 4]
```diff
@@ -0,0 +1,38 @@
+<template>
+  <div
+    class="mkr__checkbox"
+    role="checkbox"
+    :class="[{ 'checked' : value }, size]"
```
Comment (GuillaumeBAECHLER) [inline packages/mikado_reborn/src/components/Checkbox/Checkbox.vue:5]: Modifiers should follow the BEM convention.
```suggestion
    :class="[{ 'mkr__checkbox--checked' : value }, `mkr__checkbox--${size}`]"
```
Don't forget to update the sass rules according to this 😅

File: packages/mikado_reborn/src/components/index.ts [hunk 5]
```diff
@@ -11,3 +11,4 @@ export * from './Textfield';
 export * from './Textarea';
 export * from './Chips';
 export * from './Snackbar';
+export * from './Checkbox';
```
Comment (GuillaumeBAECHLER) [inline packages/mikado_reborn/src/components/index.ts:14]: Please move this below Card.
Some items are not in the right place... 😕

**Review by GuillaumeBAECHLER — CHANGES_REQUESTED** [review]
Nice thx 🙏

Some adjustments needs to be done in order to be 100% compliant with the existing codebase and the design system mockups.

- The state used for the focus should be used for the hover state too.
- No border should be present on the checkbox once checked
- Add box sizing in component styles (26px size on storybook instead of the 24px on mockups)
- Border-radius should be 4px for both sizes, thus, there is no need to redefine them.

**Comment by GuillaumeBAECHLER** [comment]
Also :

- The checkbox should emit a change event instead of an input event
 
We might encounter some difficulties while using this component. For example, as it is, we can't bind a value to the checkbox, so we won't be able to interact with arrays in the v-model.
I would suggest splitting the 'value' prop and a 'selected' prop to match most cases.

File: packages/mikado_reborn/src/components/index.ts (hunk 5 above)
Comment (hhochart29) [reply packages/mikado_reborn/src/components/index.ts:14]: 👍

File: packages/mikado_reborn/src/utils/colors.js [hunk 6]
```diff
@@ -5,6 +5,7 @@ const colors = [
   'secondary',
   'secondary-light',
   'secondary-dark',
+  'secondary-30',
```
Comment (GuillaumeBAECHLER) [inline packages/mikado_reborn/src/utils/colors.js:8]:
```suggestion
  'secondary-20',
```

File: packages/mikado_reborn/src/components/Checkbox/Checkbox.scss [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  &:focus-within, &:hover {
+    outline: none;
+    box-shadow: 0 0 0 0.5rem map-get($colors, 'secondary-20');
+    border-color: map-get($colors, 'secondary');
+  }
+
+  &--small {
+    width: 2rem;
+    height: 2rem;
+    border-radius: 0.5rem;
```
Comment (GuillaumeBAECHLER) [inline packages/mikado_reborn/src/components/Checkbox/Checkbox.scss:34]:
```suggestion
```

**Review by GuillaumeBAECHLER — APPROVED** [review]
LGTM ! 🎉👍

**Pull request merged**